cmake/OpenCVConfig.cmake

```diff
@@ -47,7 +47,7 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
+  if(BUILD_SHARED_LIBS)
```

Comment:
This makes sense, but it's backwards-incompatible, and I'm afraid it's going to break builds. I'd be okay with having this in master, but in 2.4... I'd rather not.